indexer/mailbox/src/lib.rs

```diff
@@ -0,0 +1,108 @@
+mod encode;
+mod message;
+
+extern crate alloc;
+
+use fuel_indexer_macros::indexer;
+use fuel_indexer_plugin::prelude::*;
+
+use crate::{encode::Decode, message::HyperlaneMessage};
+
+/// The log id (i.e. the value of rB in the LogData) of a dispatched message log.
+/// "hyp" in bytes
+const DISPATCHED_MESSAGE_LOG_ID: u64 = 0x687970u64;
+
+/// The contract ID of the Mailbox.
+/// See https://github.com/FuelLabs/fuel-indexer/issues/451 for a better configuration path.
+const MAILBOX_CONTRACT_ID: &str =
+    "0x968a13b3bd8310939636afa85b64bd2a533dbd770bc21f0a00e0def18c32ee2b";
+
+struct LogMetadata {
+    contract_id: Address,
+    block_number: u64,
+    block_hash: Bytes32,
+    transaction_hash: Bytes32,
+    transaction_index: u64,
+    // The index of the relevant receipt in the transaction
+    receipt_index: u64,
+}
+
+impl DispatchedMessage {
+    fn new(message: HyperlaneMessage, log_metadata: LogMetadata) -> Self {
+        let message_id = Bytes32::from(message.id().to_fixed_bytes());
+        Self {
+            // See https://fuellabs.github.io/fuel-indexer/v0.1.12/components/database/ids.html
+            id: message.nonce as u64,
+
+            version: u32::from(message.version),
+            nonce: message.nonce,
+            origin: message.origin,
+            sender: Bytes32::from(message.sender.to_fixed_bytes()),
+            destination: message.destination,
+            recipient: Bytes32::from(message.recipient.to_fixed_bytes()),
+            // Encode the message body as hex characters, and quote it.
+            // See https://github.com/FuelLabs/fuel-indexer/issues/450
+            body: Json(format!("\"{}\"", hex::encode(message.body))),
+            message_id,
+
+            // Log metadata
+            contract_id: log_metadata.contract_id,
+            block_number: log_metadata.block_number,
+            block_hash: log_metadata.block_hash,
+            transaction_hash: log_metadata.transaction_hash,
+            transaction_index: log_metadata.transaction_index,
+            receipt_index: log_metadata.receipt_index,
+        }
```

Comment:
is there no spread operator?

Reply:
unfortunately no
The closest is you can do:
```
MyStruct {
  foo,
  ..my_existing_struct
}
```
but `my_existing_struct` has to be an instance of the `MyStruct` struct